### PR TITLE
[SH4RE-20] Send "Create Listing" Entries to Backend

### PR DESCRIPTION
**Pull request by bbryanb333**

### CODE REVIEW

File: SH4RE/SH4RE/FirebaseDB.swift [hunk 1]
```diff
@@ -102,3 +103,44 @@ func collectionRead(collectionPath : String, completion: @escaping ((QuerySnapsh
     }
     return
 }
+
+extension UIImage {
```
Comment (WasifButt) [inline SH4RE/SH4RE/FirebaseDB.swift:107]: what does this method do? Can we add some comments

File: SH4RE/SH4RE/CreateListingView.swift [hunk 2]
```diff
@@ -87,7 +87,13 @@ struct CreateListingView: View {
                     .padding()
                 
                 Button(action: {
-                    print("Post tapped!")
+                    var listing_fields = ["Title": title, "Description" : description, "Price" : cost, "Category" : drop_down_selection]
+                    let document_id = documentWrite(collectionPath: "Listings",data:listing_fields)
+                    //TODO : increment images when we add ability to upload multiple
+                    let image_path = "listingimages/" + document_id + "/1.jpg"
+                    storageManager.upload(image: image, path: image_path)
```
Comment (WasifButt) [inline SH4RE/SH4RE/CreateListingView.swift:94]: where will this image be stored, is it in cloud storage and not the database? If so, is there a way to relate the image ID to the database entry?

Comment (bbryanb333) [reply SH4RE/SH4RE/CreateListingView.swift:94]: The image can't directly be stored in the database so we're instead storing it inside firebase cloud storage. The image path with upload to the database in the next line is the location of where the image is stored. So If we need the image we first get the path then request the image

File: SH4RE/SH4RE/FirebaseDB.swift (hunk 1 above)
Comment (bbryanb333) [reply SH4RE/SH4RE/FirebaseDB.swift:107]: Just allows us to resize our image for easier storage in firebase cloud storage

**Pull request merged**